storagenode/storagenodedb/database.go

```diff
@@ -87,26 +157,77 @@ func NewTest(log *zap.Logger, storageDir string) (*DB, error) {
 	}
 	pieces := filestore.New(log, piecesDir)
 
-	infodb, err := NewInfoTest()
+	versionsPath := ""
+	versionsDB, err := openTestDatabase()
 	if err != nil {
 		return nil, err
 	}
 
-	return &DB{
-		log: log,
-
+	db := &DB{
+		log:    log,
 		pieces: pieces,
-		info:   infodb,
+		kdb:    teststore.New(),
+		ndb:    teststore.New(),
+		adb:    teststore.New(),
+
+		// Initialize databases. Currently shares one info.db database file but
+		// in the future these will initialize their own database connections.
+		versions:          newVersions(versionsDB, versionsPath),
+		v0PieceInfo:       newPieceInfo(versionsDB, versionsPath),
+		bandwidth:         newBandwidth(versionsDB, versionsPath),
+		console:           newConsole(versionsDB),
+		orders:            newOrders(versionsDB, versionsPath),
+		pieceExpirationDB: newPieceExpirationDB(versionsDB, versionsPath),
+		pieceSpaceUsedDB:  newPieceSpaceUsedDB(versionsDB, versionsPath),
+		reputation:        newReputationDB(versionsDB, versionsPath),
+		storageUsage:      newStorageusageDB(versionsDB, versionsPath),
+		usedserials:       newUsedSerials(versionsDB, versionsPath),
+		vouchers:          newVouchers(versionsDB, versionsPath),
+	}
+	return db, nil
+}
+
+// openDatabase opens or creates a database at the specified path.
+func openDatabase(path string) (*sql.DB, error) {
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
+		return nil, err
+	}
+
+	db, err := sql.Open("sqlite3", "file:"+path+"?_journal=WAL&_busy_timeout=10000")
```

Comment:
why is this specific to `sqlite3`? Will/do we ever use other db types?

Reply:
Isn't this telling the sql package to use the sqlite3 driver? I think we need that to properly connect. This line was previously inside of `InfoDB` it just got moved here.